unit_tests/src/test_server_properties.cpp

```diff
@@ -56,3 +56,27 @@ TEST_CASE("server_properties is threadsafe", "[thread-safety]")
         }
     }
 }
+
+TEST_CASE("server_properties map function is threadsafe", "[thread-safety]")
+{
+    const auto limit{1000};
+
+    // Start a thread that will generally get in the way a lot by writing
+    // to the irods::server_properties object
+    std::thread persistent_write([]() {
+        for (auto i = 0; i < limit; i++) {
+            irods::set_server_property(std::to_string(i), i);
+        }
```

Comment:
Consider making the thread sleep for a few milliseconds so that there is overlap.

Reply:
Got around this by mimicking `jthread` cooperative cancelling instead.